pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,234 @@
+
+            if i == policy_params - 1 and parallel_type == "parallel_scan":
+                updated_types.inferred_types[param.name] = "bool"
+                updated_types.is_arg.add(param.name)
+
+    if len(param_list) == policy_params:
+        if not len(updated_types.inferred_types): return None
+        return updated_types
+
+    # Handle Keyword args, make sure they are treated by queuing them in args
+    if len(passed_kwargs.keys()):
+        # add value to arguments list so the value can be assessed
+        for param in param_list[policy_params:]:
+            if param.name in passed_kwargs:
+                args_list.append(passed_kwargs[param.name])
+
+    # Handling arguments other than policy args, they begin at value_idx in args list
+    value_idx: int = 3 if handled_args.name != None else 2 
+
+    assert (len(param_list) - policy_params) == len(args_list) - value_idx, f"Unannotated arguments mismatch {len(param_list) - policy_params} != {len(args_list) - value_idx}"
+
+    # At this point there must more arguments to the workunit that may not have their types annotated
+    # These parameters may also not have raw values associated in the stand alone format -> infer types from the argument list
+
+    # DataType class has all supported pk datatypes, we ignore class members starting with __
+    supported_np_dtypes = [attr for attr in dir(DataType) if not attr.startswith("__")]
+
+    for i in range(policy_params , len(param_list)):
+        param = param_list[i]
+        value = args_list[value_idx + i - policy_params]
+
+        if isinstance(value, View) and value.layout != Layout.LayoutDefault:
+            updated_types.layout_change[param.name] = "LayoutRight" if value.layout == Layout.LayoutRight else "LayoutLeft"
+
+        if param.annotation is not inspect._empty:
+            continue
+
+        param_type = type(value).__name__
+
+        # switch integer values over 31 bits (signed positive value) to pk.int64
+        if param_type == "int" and value.bit_length() > 31:
+            param_type = "numpy:int64"
+
+        # check if package name is numpy (handling numpy primitives)
+        pckg_name = type(value).__module__
+
+        if pckg_name == "numpy":
+            if param_type not in supported_np_dtypes:
+                raise TypeError(f"Numpy type {param_type} is unsupported")
+
+            # numpy:<type>, Will switch to pk.<type> in parser.fix_types
+            param_type = pckg_name +":"+ param_type
+
+        if isinstance(value, View):
+            view_dtype = get_pk_datatype(value.dtype)
+            if not view_dtype:
+                raise TypeError("Cannot infer datatype for view:", param.name)
+
+            param_type = "View"+str(len(value.shape))+"D:"+view_dtype
+
+        updated_types.inferred_types[param.name] = param_type 
+        updated_types.is_arg.add(param.name)
+
+    if not len(updated_types.inferred_types): return None
+
+    return updated_types
```

Comment:
This function is too long, it should probably be split up into multiple functions